subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/TaskInstantiator.java

```diff
@@ -38,6 +39,6 @@ public TaskInstantiator(TaskIdentityFactory taskIdentityFactory, ITaskFactory ta
 
     @Override
     public <S extends Task> S create(String name, Class<S> type) {
-        return taskFactory.create(taskIdentityFactory.create(name, type, project), NO_PARAMS);
+        return taskFactory.create(taskIdentityFactory.create(name, type, project, UserCodeSource.BY_RULE), NO_PARAMS);
```

Comment:
❓ 

Is this it?

https://docs.gradle.org/current/javadoc/org/gradle/api/NamedDomainObjectCollection.html#addRule(java.lang.String,groovy.lang.Closure)

If so, it seems all variants have a description. Can we get that description in the provenance?

Reply:
The provenance we're concerned with here is "what created/registered the task".  This is so that upon failure, we can quickly navigate to that location and get more info about the task's configuration, etc.

This case here is covering [Task Rules](https://docs.gradle.org/current/userguide/controlling_task_execution.html#sec:task_rules) and isn't the common case.

So I'm not sure what else we could do here other than say a rule created this task...sorry, we can't be more specific.